### DISCUSSION TITLE
怎么通过上级路由DHCP分配IP

### DISCUSSION THREAD

**Discussion by langziyang** [istoreos/istoreos#1374] [Q&A]
上级路由拨号并且分配了固定ip,怎么设置istoreos通过上级路由来分配，我关闭lan口的dhcp后还需要配置哪里

**Comment by jjm2473** [comment]
你是不是完全不了解dhcp的工作原理？dhcp只能给同一个交换机内的设备分配ip，如果你要主路由分配ip，那就只能把网络结构打平，不要用二级路由，而是采用旁路由

**Reply by langziyang** [reply]
确实对这个不太懂，非专业。那么请教一下，有没有可能把设备设置为类似交换机这种呢？然后里面的其它功能比如说openclash还是生效可用的

**Reply by jjm2473** [reply]
就是旁路由啊，后台的网络向导就可以配置。但是旁路由的情况下，客户端网关要选择到底是旁路由还是主路由。使用本地化网关可以对不同客户端指定不同网关 https://github.com/istoreos/istoreos/discussions/1281#discussioncomment-8177283 

不过，你的网络知识实在太匮乏，不建议改网络架构，改了可能又会遇到其他问题，建议你还不如在软路由的lan口加个wifi路由器，需要翻墙的就连这个wifi，比折腾旁路由配置简单多了。

**Reply by lgwylyc** [reply]
不太清楚你的具体要做什么？实际上可能没人这么干。不过就字面需求是可以实现的，需要有1台支持DHCP relay功能的3层交换机，上级路由作为DHCP服务器，上级路由和PC处于不同vlan，istoreos作为旁挂设备，需要用就把PC的网关指过去就行了。
